spyro/solvers/time_integration_central_difference.py

```diff
@@ -44,11 +44,16 @@ def central_difference(wave, source_id=0):
     usol_recv = []
     save_step = 0
     for step in range(nt):
-        rhs_forcing.assign(0.0)
+        # Basic way of applying sources
+        wave.update_source_expression(t)
```

Comment:
This method needs to just pass when MMS isn't used. Assigning something takes a while, it is really significant in 3D.

Reply:
Nevermind, I saw that it just passes when MMS isn't used in the code down below